app/controllers/application_controller.rb

```diff
@@ -46,10 +50,10 @@ def authentication_failed
     end
   end
 
-  def resource_not_found_handler(exception)
+  def generic_exception_handler(exception)
     respond_to do |format|
       format.json do
-        render json: {status: 404, reason: exception.to_s}, status: 404
+        render json: {status: exception.status, reason: exception.to_s}, status: exception.status
       end
       format.html do
```

Comment:
Przekierowanie na index dla HTML nie ma sensu w tej usłudze. Ale rozumiem, że to nie Twój kod, więc jak nie zapomnę, to poruszę to na następnym spotkaniu.